### ISSUE TITLE
Keras LSTM:  how to compute symbolic gradients

### ISSUE THREAD

**Issue by shaifugpt** [keras-team/keras#13030]
I want to start by taking an example for a normal neural network. Let the weights between input and hidden nodes are W_ij and weights between hidden nodes and output are W_jk. The  gradient of error wrt W_ij= W_jk * g'(W_ij * x) *x and wrt W_jk =g( W_ij * x). This way how can we compute the symbolic gradients for weights in keras for LSTM. Tensorflow gradients function only returns this: tf.gradients(model1.output,model1.trainable_weights[0]).

**Comment by dabasajay** [comment]
There are 3 popular methods to calculate the derivative/gradients:
1. Numerical differentiation
2. Symbolic differentiation
3. Automatic differentiation
Tensorflow (and thus Keras) uses automatic differentiation and hence you get back the value of gradients and not the symbolic expression.
If you want mathematical expression, maybe use something like Matlab
Let me know if I misunderstood your question or need to add something
